extensions/dns-srv.md

```diff
@@ -38,6 +38,10 @@ Example:
 
     _ircs._tcp SRV 0 1 6697 irc.example.org.
 
+After discovering an IRC server via SRV records, the client MUST check that the
+TLS server certificate is valid for the original hostname (`example.org` in the
+example above).
```

Comment:
This is not very practical when `example.org` hosts other services than the IRC network. If the private certificate of IRC server is leaked, TLS communications of other services are compromised as well.

Reply:
I don't understand the concern here. Is the claim that IRC servers have a larger attack surface for compromises than, say, the HTTP servers or load balancers that would normally serve a certificate covering `example.org`?

Reply:
Yes, attack surface of IRC server is much larger than of reverse proxy HTTP server.

Reply:
I don't agree, but an operator who felt this way could put a TLS-terminating reverse proxy in front of their IRC server. (For example, nginx supports this with the [stream](https://docs.nginx.com/nginx/admin-guide/security-controls/terminating-ssl-tcp/) directive.)

Reply:
It could indeed be a good solution.

Reply:
> This is not very practical when `example.org` hosts other services than the IRC network. If the private certificate of IRC server is leaked, TLS communications of other services are compromised as well.

As explained above; if you do not enforce that the certificate is valid for the original hostname (the one the client looked up), then a trivial DNS MITM is possible, to direct the client to a hostname of the attacker's choosing, defeating the entire point of using TLS.

Reply:
> This is not very practical when `example.org` hosts other services than the IRC network.

It seems to be practical enough for XMPP server operators. XMPP also requires the certificate to validate against the domain from JID, not against the SRV target.

Reply:
Just throwing an idea up there (haven't given it much thought). Could the TLS hostname validation use the SRV hostname instead (for exampole `_ircs._tcp.example.org`). That way we would be restricting the certificate for only use with the irc service and not any other service running in the `example.org` domain (where comprimise of cert could allow it to be used for other services in the root domain).

Reply:
While you can't have underscores in dNSName, RFC 4985 exists as an alternative. Sadly, Baseline Requirements of CA/B Forum don't allow issuance of SRVName yet, but there is a [discussion](https://github.com/cabforum/servercert/issues/268) to allow them.